src/web/demo.rkt

```diff
@@ -328,19 +328,58 @@
        (when (eof-object? formula)
          (raise-herbie-error "no formula specified"))
        (parse-test formula)
-       (define hash (sha1 (open-input-string formula-str)))
+       (define hash (sha1 (open-input-string (~s 'improve formula (get-seed)))))
        (body hash formula))]
     [_
      (response/error "Demo Error"
                      `(p "You didn't specify a formula (or you specified several). "
                          "Please " (a ([href ,go-back]) "go back") " and try again."))]))
 
+(define (improve-start-extract post-data)
+  (define form-str (hash-ref post-data 'formula))
+  (define form-seed (hash-ref post-data 'seed))
+  (list form-str form-seed))
+
+(define (improve-start-helper req body go-back)
+  (define post-body (request-post-data/raw req))
+  (define post-data (cond (post-body (bytes->jsexpr post-body)) (#t #f)))
+  (match (improve-start-extract post-data)
+    [(list formula-str seed*)
+     (define formula
+       (with-handlers ([exn:fail? (λ (e) #f)])
+         (read-syntax 'web (open-input-string formula-str))))
+     (unless formula
+      (raise-herbie-error "bad input: did you include special characters like `#`?"))
```

Comment:
Why not just raise the `herbie-error` in the `exn:fail?` handler? Also why not move this to a separate method?